test/login.test.ts

```diff
@@ -0,0 +1,41 @@
+import { chromium, Page, Browser, BrowserContext } from "playwright"
+
+// NOTE: this is hard-coded and passed as an environment variable
+// See the test job in ci.yml
+const PASSWORD = "e45432jklfdsab"
+
+describe("login", () => {
+  let browser: Browser
+  let page: Page
+  let context: BrowserContext
+
+  beforeAll(async () => {
+    browser = await chromium.launch()
+    context = await browser.newContext()
+  })
+
+  afterAll(async () => {
+    await browser.close()
+  })
+
+  beforeEach(async () => {
+    page = await context.newPage()
+  })
+
+  afterEach(async () => {
+    await page.close()
+    // Remove password from local storage
+    await context.clearCookies()
+  })
+
+  it("should be able to login with the password from config.yml", async () => {
+    await page.goto("http://localhost:8080")
```

Comment:
Let's do something like:

```suggestion
    await page.goto(process.env.CODE_SERVER_ADDRESS)
```
And then set that in the `ci.yaml`. We could still default to `http://localhost:8080` though. We could add a little `getCodeServerAddress()` utility function or something since it'll be used by multiple tests.